### ISSUE TITLE
Additional space in 0d array repr

### ISSUE THREAD

**Issue by lesteve** [numpy/numpy#9185]
Snippet:
```py
import numpy as np
print(repr(np.array(1.)))
```

Output on 1.12.1:
```
array(1.0)
```

Output on master:
```
array( 1.0)
```
Note the space before `1.0`.

git bisect gives that the commit introducing the change is d2b06fe879f5b2b14de3dad0f517561a0c815df0.

Probably not such a big deal, but I guess it was not intentional. We noticed because it was breaking one of our doctest on scikit-learn.

**Comment by lesteve** [comment]
FWIW 1.13.0rc2 is not affected by this.

**Comment by rkern** [comment]
I believe that #9139 is intended to fix this.

**Comment by lesteve** [comment]
Good point, I search for an existing issue but I missed that. Looking at #9139 it will indeed fix this issue while at the same break more of our doctests. We can probably work around this in some way or another though.

**Comment by lesteve** [comment]
@ahaldane @eric-wieser @charris this one can be closed now that #9139 has been merged.

**Comment by ahaldane** [comment]
Thanks, closed.
